ADDING_RESOURCES.md

```diff
@@ -148,4 +148,7 @@ for more details.
 
 ### Azure Specifics
 
+When deleting an Azure resource, `DeleteAzureControlledResourceStep.java` contains a base class for the deletion steps
```

Comment:
This should be more prescriptive, something like "if you're implementing a new azure resource, you must subclass `DeleteAzureControlledResourceStep` for your resource deletion step".